### ISSUE TITLE
MOON-49 - Create the breadcrumb component

### ISSUE THREAD

**Issue by jahia-ci** [Jahia/moonstone#897]
> Imported from Jira, on Thu Jan 30, 2025
> Issue: [MOON-49](https://jira.jahia.org/browse/MOON-49) in project: Moonstone
> Priority: Minor Type: Story
> Reporter: @ffffffelix (Félix Vallé)
> Assignee: @jahia-ci (Thomas Draier)
> Created: Mon Jan 27, 2020, closed: Wed Mar 11, 2020
> Status: Closed (resolution: Done)
> Sprint: Code Runners - Sprint 34 started on Mon Mar 2, 2020 completed on Mon Mar 16, 2020
> **Parent Epic**: [BACKLOG-11608 - jContent - Header](https://jira.jahia.org/browse/BACKLOG-11608) (Closed)
>
> **Links**:
> - depends on: [DESIGN-566 - Define the breadcrumbs component](https://jira.jahia.org/browse/DESIGN-566) (Closed)
> - depends on: [MOON-50 - Create Chip component](https://jira.jahia.org/browse/MOON-50) (Closed)
> - depends on: [MOON-51 - Update Button component](https://jira.jahia.org/browse/MOON-51) (Closed)
> - is depended on by: [BACKLOG-11608 - jContent - Header](https://jira.jahia.org/browse/BACKLOG-11608)
> - is depended on by: [BACKLOG-12325 - Integrate breadcrumb component](https://jira.jahia.org/browse/BACKLOG-12325)
>

Start with the API (Name of the component and properties), then:
* implement the component
  * Buttons with seperators (icon ">" or "/" to be clarified by @jahia-ci (Pascal Chirol <X>))
  * Chip is not part of this component (to be added during integration in the header)
  * Check if we need Selenium attributes at this level (or in the integration story)
* Storybook
* Unit Test

We need a CSS ellipsis when the name is too long (needs UX guidance...)
* Check how it was done in CMM

 <https://www.figma.com/file/939bW74C3TLW5VAzK23uox/moonstone-components?node-id=2072%3A17546>

For Selenium the following data-roles existed:

To get each linked element of the breadcrumbs: 
By.xpath("//**<@data-cm-role='breadcrumb'>//button//**<@data-cm-role='breadcrumb-name'>")

Hidden items locator (3 dots):
 By.xpath("//*<@data-sel-role='hidden-items'>")

Hidden-items menu (expanding after click on 3 dots):
 By.xpath("//div<@data-sel-role='hidden-menu'>//li")

**Comment by jahia-ci** [comment]
> Comment by: @jahia-ci (Pascal Chirol [X]) on Mon Mar 2, 2020 _edited_


*@rknj (Michael De Checchi)* 
*Buttons with seperators (icon ">" or "/" to be clarified by_ @jahia-ci (Pascal Chirol <X>)_)*
 **MOOSTONE DESIGN SYSTEM components are the TRUST.** (v8 figma mockups are functional visions more that final components)

**Comment by jahia-ci** [comment]
> Comment by: @rknj (Michael De Checchi) on Mon Mar 2, 2020 _edited_


OK so we follow the design of the component itself (see the link in the description)

**Comment by jahia-ci** [comment]
> Comment by: @sdusolle (Sonia Dusolle) on Mon Mar 9, 2020 _edited_


Tested on local storybook
* breadcrumb is matching design in Figma
* separator is ">"
* items correspond to ghost button in small

--> one issue: long names are not cropped, we miss the CSS ellipsis

**Comment by jahia-ci** [comment]
> Comment by: @sdusolle (Sonia Dusolle) on Wed Mar 11, 2020 _edited_


Examples of long labels have been added to the storybook
